cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ResponseUtils.java

```diff
@@ -35,12 +36,22 @@
   private ResponseUtils() {
   }
 
-  static void setResponseCode(HttpServletResponse response, int code, boolean json) {
+  static void setResponseCode(HttpServletResponse response, int code, boolean json, KafkaCruiseControlConfig config) {
```

Comment:
**Proposal:** Rather than passing a `KafkaCruiseControlConfig`, would it make sense to pass `String httpCorsExposeHeaders` parameter, which is `null` if `cors` is disabled?

**Issue:** Then this enables a caller to pass the necessary `String` for `Access-Control-Expose-Headers` in case `cors` is enabled. Currently `AbstractCruiseControlResponse#writeSuccessResponse` calls this function with `null` config, which prevents adding this tag to the header even when `cors` is enabled.

**Details:** If we apply the proposed solution above, then we can initialize `_httpCorsExposeHeaders` in a constructor of `AbstractCruiseControlResponse` with one argument, which is called by the constructor of each class that extends from it.

Reply:
i might need access to few more headers here, so `config` is used instead of a simple `string`. 
Also, we don't need to send the `Access-Control-Expose-Headers` for error responses as we don't have `Task-Id` defined in that case. so the choice is made accordingly.

Reply:
Thanks for the clarification regarding the error responses!

* Do we want to send the `Access-Control-Expose-Headers` in `writeSuccessResponse` of `AbstractCruiseControlResponse`? This function is called to return successful response of any CC endpoint. Internally, this function calls `setResponseCode(response, SC_OK, json, null);`, which does not pass `config`. 

^ (Just to double check my understanding) Would you confirm that this will be addressed in a later patch -- i.e. not returning `Access-Control-Expose-Headers` for success case regardless of `cors` enabled or not.

Reply:
Yes, i will submit another patch which will restrict the usage of exposing CORS headers.